scripts/apps/authoring-react/authoring-react.tsx

```diff
@@ -560,13 +584,21 @@ export class AuthoringReact<T extends IBaseRestApiResponse> extends React.PureCo
                 proofReadingTheme,
             );
 
-            this.props.onEditingStart?.(initialState.itemWithChanges);
-
             this.setState(initialState);
 
             if (this.componentRef != null) {
                 this.cleanupFunctionsToRunBeforeUnmounting.push(focusFirstChildInput(this.componentRef).cancel);
             }
+
+            return initialState;
+        });
+    }
+
+    componentDidMount() {
+        const {authoringStorage} = this.props;
+
+        this.switchProfile(() => authoringStorage.getEntity(this.props.itemId)).then((initialState) => {
```

Comment:
since `switchProfile` profile already provides an ID of an item, you should pass that to `getEntity` instead of the one from props.